docs/articles/commands/custom_context_checks.md

```diff
@@ -73,4 +77,18 @@ Then we use the check like such:
 [Command("dm")]
 [DirectMessageUsage(DirectMessageUsage.RequireDMs)]
 public async ValueTask RequireDMs(CommandContext commandContext) => await commandContext.RespondAsync("This command was executed in a DM!");
-```
\ No newline at end of file
+```
+
+## Advanced Features
+
+The classes you use to implement checks participate in dependency injection, and you can request any type you previously supplied to the service provider in a public constructor. Useful applications include, but are not limited to, logging or tracking how often a command executes.
```

Comment:
"Context checks do support constructor dependency injection, which allows for easy logging or metric tracking"

Reply:
i specifically wanted the "include, but are not limited to" part, so as to not make it sound that's the main use-case - the sky's the limit